hazelcast/src/main/java/com/hazelcast/internal/management/request/ChangeClusterStateRequest.java

```diff
@@ -0,0 +1,64 @@
+package com.hazelcast.internal.management.request;
+
+import com.eclipsesource.json.JsonObject;
+import com.hazelcast.cluster.ClusterState;
+import com.hazelcast.internal.management.ManagementCenterService;
+
+import java.io.IOException;
+
+import static com.hazelcast.util.JsonUtil.getString;
+
+public class ChangeClusterStateRequest implements AsyncConsoleRequest {
+
+    private String state;
+
+    public ChangeClusterStateRequest() {
+    }
+
+    public ChangeClusterStateRequest(String state) {
+        this.state = state;
+    }
+
+    @Override
+    public int getType() {
+        return ConsoleRequestConstants.REQUEST_TYPE_CHANGE_CLUSTER_STATE;
+    }
+
+    @Override
+    public Object readResponse(JsonObject in) throws IOException {
+        return getString(in, "result", "FAIL");
+    }
+
+    @Override
+    public void writeResponse(ManagementCenterService mcs, JsonObject out) throws Exception {
+        JsonObject result = new JsonObject();
+        String resultString = "SUCCESS";
+        try {
+            if (state.equalsIgnoreCase("ACTIVE")) {
+                mcs.getHazelcastInstance().getCluster().changeClusterState(ClusterState.ACTIVE);
+
+            } else if (state.equalsIgnoreCase("FROZEN")) {
+                mcs.getHazelcastInstance().getCluster().changeClusterState(ClusterState.FROZEN);
+
+            } else if (state.equalsIgnoreCase("PASSIVE")) {
+                mcs.getHazelcastInstance().getCluster().changeClusterState(ClusterState.PASSIVE);
+            }
```

Comment:
Yes. However, that case is not possible under normal circumstances since it comes directly from the UI.

Only way is to someone spoof the call to Management Center server somehow, which ends in a failure message.